pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -68,7 +68,7 @@ type Manager interface {
 
 	// GetTopologyHints implements the Topology Manager Interface and is
 	// consulted to make Topology aware resource alignments
-	GetTopologyHints(pod v1.Pod, container v1.Container) map[string][]topologymanager.TopologyHint
```

Comment:
Hmm. The reason I didn't include this in the original PR after our discussion, is because it was unclear to me how to return an error on a per-resource basis. I'm not sure how I feel about returning a single error for the entire call, if just one of the resources is in error.

Reply:
can you explain why it is important to be in  per resource basis. As I see it  if one resource failed then we can't admit the pod, right?

Reply:
@lmdaly made the point that maybe we want to wait and let the allocation attempt fail the pod admission instead of the topology hint generation failing it. I'm still on the fence about it, but I think it warrants further discussion.

Also, I'd like to see all such resources that are failing this way and not short circuit on the first one if possible.

Reply:
There might be a case were a resource is healthy and node got selected then it become unhealthy (no topology hint) healthy  again allocate device (not sure if it can happen), but I would like to avoid such race and block it on the topology hint generation. 
Regarding the other point I will update the PR to see all resource once we agree that this is a good direction.